### ISSUE TITLE
YamlFileLoader does 'supports' based purely on file extension

### ISSUE THREAD

**Issue by salvocanna** [symfony/symfony#20308]
After playing a while with the symfony configuration I realised there's no way to make symfony parse and load a yaml file if its extension is not in ['yml', 'yaml']

Shouldn't it be a configuration? 

Something like:

```
imports:
    - { resource: my_custom_config.ext, format: yml }
```

I can submit a pull request

I'm sorry if this may be already implemented somehow and I'm not aware of it.

Cheers

**Comment by Ener-Getick** [comment]
Extending the `supports` method of the yaml loader should be enough, right?

**Comment by chalasr** [comment]
We should not have to add anything as there is already a `type` key e.g:

```
imports:
    - { resource: my_custom_config.ext, type: yml }
```

Unfortunately, this key is not used at all in the DI `YamlFileLoader`. The `supports()` method looks like:

```php
public function supports($resource, $type = null)
{
    return is_string($resxource) && in_array(pathinfo($resource, PATHINFO_EXTENSION), array('yml', 'yaml'), true));
}
```

Imho we should change it to something like:

```php
public function supports($resource, $type = null)
{
    return is_string($resxource) && (in_array(pathinfo($resource, PATHINFO_EXTENSION), array('yml', 'yaml'), true) || in_array($type, array('yml', 'yaml'), true));
}
```

So we don't care about the resource extension except if `type` is not provided, same for other file loaders. WDYT?

**Comment by salvocanna** [comment]
@chalasr exactly also, in the same loader, `parseImports($content, $file)`:

```php
$this->import($import['resource'], null, isset($import['ignore_errors']) ? (bool) $import['ignore_errors'] : false, $file);
``` 
should become something like

```php
$this->import($import['resource'], isset($import['type']) ? (bool) $import['type'] : null, isset($import['ignore_errors']) ? (bool) $import['ignore_errors'] : false, $file);
```

as the second parameter is the type.

I think that's all.

**Comment by chalasr** [comment]
Good catch @salvocanna, shall I propose the change or do you want to (that's your issue, you deserve it)?

**Comment by salvocanna** [comment]
@chalasr I'd like to do it even if it's the first one ever made (if I'm successful..)

Which branch should it go to?

I currently use symfony 2.8 so would be nice to see it there, but depends, should it be seen as a feature or bug fix?

**Comment by Ener-Getick** [comment]
I'd go for a feature as it never worked before but others may think otherwise.

**Comment by chalasr** [comment]
@salvocanna I would propose it as a bugfix, as the type key exists and does not look useful as is. At worst your PR can be merged on another branch than the one you target at first (you can also change the base yourself from github if someone ask it). 

So if you go for a bugfix then 2.7 should be targeted, see http://symfony.com/doc/current/contributing/code/patches.html#step-3-submit-your-patch for details and don't forget to update/add the corresponding tests :)

**Comment by ogizanagi** [comment]
Just to mention the [Routing\Laoder\YamlFileLoader](https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Routing/Loader/YamlFileLoader.php#L95-L98) only check for the `yaml` type (which makes sense, it's the targeted loader, not the file extension):

```php
public function supports($resource, $type = null)
{
    return is_string($resource) && in_array(pathinfo($resource, PATHINFO_EXTENSION), array('yml', 'yaml'), true) && (!$type || 'yaml' === $type);
}
```

It also give precedence to the `type` if explicitly set.

**Comment by ogizanagi** [comment]
See https://github.com/symfony/symfony/pull/20611
